Warp.WebApp/Pages/Preview.cshtml.cs

```diff
@@ -0,0 +1,98 @@
+using CSharpFunctionalExtensions;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Claims;
+using Warp.WebApp.Helpers;
+using Warp.WebApp.Models;
+using Warp.WebApp.Pages.Shared.Components;
+using Warp.WebApp.Services;
+using Warp.WebApp.Services.Entries;
+
+namespace Warp.WebApp.Pages
+{
+    public class PreviewModel : BasePageModel
+    {
+        public PreviewModel(ILoggerFactory loggerFactory, IEntryService previewEntryService) : base(loggerFactory)
+        {
+            _entryService = previewEntryService;
+        }
+
+        public async Task<IActionResult> OnGet(string id, CancellationToken cancellationToken)
+        {
+            var decodedId = IdCoder.Decode(id);
+            if (decodedId == Guid.Empty)
```

Comment:
Probably, we have to rewrite `IdCoder` with usage of `CSharpFunctionalExtensions`